website/src/app/blog/how-dns-works-in-firezone/readme.mdx

```diff
@@ -1,18 +1,383 @@
+the problem DNS was originally designed to solve.
+
+DNS was created in the early 1980s to address a classic scalability problem with
+the early ARPANET.
+
+<Image
+  src="/images/blog/how-dns-based-traffic-routing-works/arpanet-logical-map.png"
+  alt="Early ARPANET logical map"
+  width={1200}
+  height={1200}
+  className="mx-auto rounded shadow"
+/>
+
+<p className="text-sm italic text-center mx-auto">
+  Figure 1: Early ARPANET logical map
+</p>
+
+The ARPANET, the precursor to the modern internet, was a network of computers
+that connected various research institutions and government agencies across the
+United States. It was the first network to use the new TCP/IP protocol suite,
+which provided an IP address to each host on the network.
+
+To connect to a host on the ARPANET, you needed its IP address. But there was a
+problem: IP addresses are hard to remember, and it turns out humans aren't great
+at remembering long strings of numbers. To solve this, a simple system was
+created to map each host's IP address to a human-friendly "hostname".
+
+Simple enough, right? Well, not quite. The system was perhaps a bit _too_
+simple.
+
+### Files don't scale
+
+Hostname to IP address mappings were stored in a single file, aptly named
+`HOSTS.TXT`, and stored on the `SRI-NIC` server -- a system maintained by the
+Network Information Center (NIC) at the Stanford Research Institute (SRI).
+
+<p className="text-sm italic text-center mx-auto">
+  Figure 2: Artist's rendition of the original HOSTS.TXT file
+</p>
+
+Whenever an ARPANET member organization (yes, you had to be a member at that
+time) wanted to add a host to the file or update its IP address, it had to
+contact the SRI (during standard business hours!) and file a formal request
+containing the updated mapping.
+
+Need to make an update over the weekend or on a US holiday? Too bad, you were
+out of luck! The system worked ok when there were only a handful of hosts to
+maintain. But, as you might imagine, it became increasingly unmanageable as more
+hosts joined.
+
+## Enter the Domain Name System (DNS)
+
+Over the course of several meetings, early ARPANET engineers (who would later go
+on to form the [IETF](https://www.ietf.org/)) devised a clever solution to the
+scalability problem they were facing: instead of storing the hostname lookup
+table in a single file, they would chop it up instead and distribute the entries
+across several "Nameservers", each one responsible for maintaining the hosts
+that belonged to a particular network, or "Domain".
+
+Hence the name: the Domain Name System, or DNS for short.
+
+### How DNS works
+
+At a high level, DNS is a heirarchical system that distributes the
+responsibility of resolving a a fully-qualified domain name (FQDN) to a series
+of nameservers, each one responsible for resolving a different part.
+
+<p className="text-sm italic text-center mx-auto">Figure 3: How DNS works</p>
```

Comment:
Will there be an actual figure? I think this would be a great addition (even though it isn't part of the core of the post.)

Reply:
Yeah still working on the graphics, this will be a Firezone-themed diagram

Reply:
I do worry though that the intro about all the DNS is a bit long. will try to make that more concise.

Reply:
> I do worry though that the intro about all the DNS is a bit long. will try to make that more concise.

I thought the same actually. Maybe say something like, "if you are familiar with how DNS works, jump ahead to xyz"?

Or can we skip over everything after the stub resolver?